### ISSUE TITLE
Suggestion: create a GitHub release after pushing each tag

### ISSUE THREAD

**Issue by aantron** [libuv/libuv#2376]
This allows GitHub to notify GitHub users of releases, if they are watching the repo for *Releases only* (at least, as I understand it).

The release text can be empty. Another easy option is to paste in the tag message.

My apologies if this has been suggested in the past; I searched only this repo.

**Comment by cjihrig** [comment]
Are these - https://github.com/libuv/libuv/releases - not getting the job done?

**Comment by aantron** [comment]
I don't think so, though it's possible that I'm not receiving emails for some other reason. I'm under the impression that the emails are only sent after using "Draft a new release."

**Comment by cjihrig** [comment]
If releases are created, but notifications aren't sent out, that sounds like a bug on GitHub's side. I'm hesitant to introduce more manual steps into the release process.

**Comment by saghul** [comment]
As someone who did releases in the past, I agree with @cjihrig here.

**Comment by aantron** [comment]
I'm speculating, but I don't think those are GitHub release objects, they are just tags that GitHub is displaying on the releases page due to the lack of GitHub release objects. Compare these two pages:

https://github.com/libuv/libuv/releases/v1.30.1
https://github.com/ocsigen/lwt/releases/4.2.1

The libuv "release" says "cjihrig *tagged*," while the Lwt "release" says "aantron *released*." Note also that on the libuv object, the "Tags" tab is highlighted, but on the Lwt object, "Releases" is highlighted.

The difference is that the Lwt release was created using "Draft a new release," which allows to create a release for a tag. I think this is what is necessary to get GitHub to send an email.

And here is a releases list from some project of mine, in which I didn't do GitHub releases at first, but then started: https://github.com/aantron/better-enums/releases/0.11.2. There, you can see the difference between what GitHub considers a release (the most recent two objects), and a tag that doesn't have a GitHub release associated with it.

**Comment by aantron** [comment]
This brief page seems to confirm: https://help.github.com/en/articles/about-releases

**Comment by cjihrig** [comment]
@aantron I think you're right. It looks like this is a long standing source of confusion: https://github.community/t5/How-to-use-Git-and-GitHub/Tag-without-release/td-p/6255.

**Comment by aantron** [comment]
Yes, I'm also learning a lot more about the subtleties now, from discussing this issue :)

**Comment by aantron** [comment]
...perhaps the most immediately relevant of which would be that I think it's best not to go back and create releases for existing tags, but only to start with the next tag (if you choose to do it, of course).
